features/full_tests/trimming.feature

```diff
@@ -0,0 +1,77 @@
+    And I run "MetadataStringsTooLargeScenario" and relaunch the crashed app
+    And I configure Bugsnag for "MetadataStringsTooLargeScenario"
+    Then I wait to receive an error
+    And the error is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+    And the exception "message" equals "Empty list doesn't contain element at index 0."
+    And the event "metaData.custom.foo" equals "abcdefghijklmnopqrstuvwxyzabcdefghijklm***<1> CHAR TRUNCATED***"
+    And the breadcrumb named "test" has "metaData.a" equal to "abcdefghijklmnopqrstuvwxyzabcdefghijklm***<1> CHAR TRUNCATED***"
+    And the event "usage.system.stringsTruncated" equals 2
+    And the event "usage.system.stringCharsTruncated" equals 2
+
+  Scenario: 1 metadata char truncated, native exception
+    When I configure the app to run in the "native, 39, abcdefghijklmnopqrstuvwxyzabcdefghijklmn" state
+    And I run "MetadataStringsTooLargeScenario" and relaunch the crashed app
+    And I configure Bugsnag for "MetadataStringsTooLargeScenario"
+    Then I wait to receive an error
+    And the error is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+    And the exception "message" equals "MetadataStringsTooLargeScenario"
+    And the event "metaData.custom.foo" equals "abcdefghijklmnopqrstuvwxyzabcdefghijklm***<1> CHAR TRUNCATED***"
+    And the breadcrumb named "test" has "metaData.a" equal to "abcdefghijklmnopqrstuvwxyzabcdefghijklm***<1> CHAR TRUNCATED***"
+    And the event "usage.system.stringsTruncated" equals 2
+    And the event "usage.system.stringCharsTruncated" equals 2
+
+  Scenario: 2 metadata chars truncated
+    When I configure the app to run in the "handled, 38, abcdefghijklmnopqrstuvwxyzabcdefghijklmn" state
+    And I run "MetadataStringsTooLargeScenario"
+    Then I wait to receive an error
+    And the error is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+    And the exception "message" equals "MetadataStringsTooLargeScenario"
+    And the event "metaData.custom.foo" equals "abcdefghijklmnopqrstuvwxyzabcdefghijkl***<2> CHARS TRUNCATED***"
+    And the breadcrumb named "test" has "metaData.a" equal to "abcdefghijklmnopqrstuvwxyzabcdefghijkl***<2> CHARS TRUNCATED***"
+    And the event "usage.system.stringsTruncated" equals 2
+    And the event "usage.system.stringCharsTruncated" equals 4
+
+  # This scenario may flake if the payload structure changes significantly because it relies upon overflowing
+  # the max payload size by a certain amount. If it fails after a structural change, verify
+  # maze_output/failed/Payload_is_too_big_by_3_breadcrumbs/errors.log to make sure it's behaving as expected,
+  # and then modify "When I configure the app to run in the "100" state" to a number that generates 3 breadcrumbs
+  # worth of data too much.
```

Comment:
I was going to say please don't add scenarios that are known to be flaky, but having read it more I think you really mean "start to fail consistently" rather than flake here?

Reply:
...although having said that isn't it likely to flake due to things like network and other breadcrumbs being slight vary in nature?  Can we turn off any breadcrumb types like to be sure that we get exactly the right number of breadcrumbs?